src/main/java/org/tomitribe/auth/signatures/SigningAlgorithm.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.tomitribe.auth.signatures;
+
+import java.util.Map;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Arrays;
+
+/**
+ * The algorithm parameter contains the name of the signature's Algorithm,
+ * as registered in the HTTP Signature Algorithms Registry defined by this document.
+ * 
+ * The signature verification is based on the signature's algorithm from the keyId
+ * parameter rather than from this algorithm.
+ * If algorithm is provided and differs from or is incompatible with the algorithm
+ * or key material identified by keyId (for example, algorithm has a value of
+ * rsa-sha256 but keyId identifies an EdDSA key), then a verification exception is
+ * raised.
+ * 
+ * The default value for this parameter should be "hs2019".
+ * 
+ * @see https://www.ietf.org/id/draft-ietf-httpbis-message-signatures-00.html
+ */
+public enum SigningAlgorithm {
```

Comment:
I keep wondering whether there's a way to have one enum as opposed to two. I'm not suggesting that we change this, more just thinking out loud.

Reply:
I've been wondering the same thing for python and golang, but I think in the draft they have decoupled the signature algorithm identifier from the details of the cryptographic algorithms, so it seems inevitable.